lib/aws_command.rb

```diff
@@ -23,12 +23,21 @@ def launch()
 
         # Expand all of the instance names so that we have a complete array
         names = []
-        options[:count].times { names << "#{options[:env]}-#{options[:type]}-#{SecureRandom.hex(5)}" }
+        minion_indexes = []
+        indexes_by_name = Hash.new
+        options[:count].times do |index|
+          name = "#{options[:env]}-#{options[:type]}-#{SecureRandom.hex(5)}"
+          names << name
+          minion_indexes << index
```

Comment:
minion_indexes seems redundant, since you could get the index of the oo_new_inst_names array.

Though, I don't think this is the behavior that you want. From what I can tell, this would only work when launching minions the first time. If you attempt to add additional minions at a later time, then the index values would start from 0 again.

What you probably want to do is to make sure we are enabling fact caching for the existing hosts (if any) and then query the hosts for the existing indexes to get the next index to use. While not ideal, since we could have hosts that are offline and not cached that would not return a value, but it is an improvement. We'll probably want to add an issue to track finding a better solution than relying on fact caching to make sure that we are not duplicating subnets (I suspect the long term solution will be to either store it in etcd or provide a way for an openshift-master to provide the needed value through an api call/utility).

Also, I think the name indexes is a bit confusing, it should be something more indicative of what it is (subnet id, network id, etc).